src/Sarif.Multitool/RewriteCommand.cs

```diff
@@ -39,6 +41,23 @@ public static int Run(RewriteOptions rewriteOptions)
             return 0;
         }
 
+        private static IDictionary<string, Uri> ConstructUriBaseIds(IEnumerable<string> uriBaseIds)
```

Comment:
>ConstructUriBaseIds [](start = 48, length = 19)

Since you defined the multi-valued `UriBaseIds` command line option in the driver, I suggest you put the code to parse it there as well, rather than restricting it to a single verb in a single tool.

Reply:
In fact it could even be an extension method on `CommonOptionsBase`, so you could say
```
IDictionary<string, Uri> originalUriBaseIds = rewriteOptions.ParseUriBaseIdsOption();
```
It could do the null check if the argument is absent, just as you do here.


---
In reply to: [214520378](https://github.com/Microsoft/sarif-sdk/pull/1062#discussion_r214520378) [](ancestors = 214520378)